### ISSUE TITLE
Figure out shared folders

### ISSUE THREAD

**Issue by ColdHeat** [ColdHeat/mech#15]
Not entirely sure if this is working or it isn't. But it would greatly increase usage if mech supported shared folders properly.

**Comment by ColdHeat** [comment]
The problem it seems is that some Vagrant VMs don't have the feature enabled by default. 

By running the following commands you should be able to turn it on. 

```
sudo apt-get install linux-headers-$(uname -r)
sudo vmware-config-tools.pl
```

Perhaps we can also explore `sudo apt-get install open-vm-tools`

**Comment by ColdHeat** [comment]
Also there are problems if the VM is unable to be connected to because it is missing a network adapter.

**Comment by ColdHeat** [comment]
It seems like vmrun is not properly linking the current directory to the vmx. Once it gets set, it doesn't change. Looks like we might need to rewrite the vmx here again as well.

**Comment by arizvisa** [comment]
does `vmrun enableSharedFolders` and `vmrun disableSharedFolders` not work for you if the vmware tools are installed? or is this referring to something else?

**Comment by ColdHeat** [comment]
I am not sure, those commands should only work if VMWare Tools is installed. There aren't a lot of things besides start and stop that will work if Tools aren't installed.

**Comment by arizvisa** [comment]
Yeah, I think vagrant expects you to have VMware tools installed in order to provide support for shared folders. So it should be an assumed requirement and not something that's actually enforced by `mech`. Most json templates for hashicorp's `packer` include the installation of VMware tools as part of the build anyways, so I think it's a safe assumption.

**Comment by thumbidea** [comment]
I just discovered Mech today. Awesome work.

I've been experimenting with shared folders. I've tried several boxes and tried reinstalling both `vmware-tools` and `open-vm-tools` with no luck. The shared folders don't get mounted when the box is brought up.

I was able to manually get shared folders working by doing:

```
$ mech init bento/ubuntu-14.04
$ mech up
$ mech ssh
> mkdir shared
> vmhgfs-fuse .host:/mech ~/shared
```

This tells me that the `vmware-tools` in the `bento/ubuntu-14.04` box are working correctly. I'm guessing that the problem is a missing mount point in the image.

BTW, `vmhgfs-fuse` requires that the mount point be owned by the user invoking it.
